contribs/gmf/src/directives/print.js

```diff
@@ -0,0 +1,803 @@
+   * @type {?angular.$q.Deferred}
+   * @private
+   */
+  this.requestCanceler_ = null;
+
+  /**
+   * @type {?angular.$q.Promise}
+   * @private
+   */
+  this.statusTimeoutPromise_ = null;
+
+  /**
+   * @type {gmf.PrintState}
+   * @export
+   */
+  this.printState = gmf.PrintState.CAPABILITIES_NOT_LOADED;
+
+  /**
+   * Current report reference id.
+   * @type {string}
+   * @private
+   */
+  this.curRef_ = '';
+
+  /**
+   * Formats availables in capabilities.
+   * @type {Array.<string>}
+   * @private
+   */
+  this.formats_ = [];
+
+  /**
+   * An array of attributes objects from capabilities.
+   * @type {Array.<Object>}
+   * @private
+   */
+  this.layouts_ = [];
+
+  /**
+   * An attributes object from capabilities.
+   * @type {Object}
+   * @private
+   */
+  this.layout_ = {};
+
+  /**
+   * @type {Array.<number>}
+   * @private
+   */
+  this.paperSize_ = [];
+
+  /**
+   * @type {gmfx.PrintFields}
+   * @export
+   */
+  this.fields = {};
+
+  /**
+   * @type {number}
+   * @export
+   */
+  this.rotation = 0;
+
+  /**
+   * @type {Array.<number>|null}
+   * @private
+   */
+  this.onDragPreviousMousePosition_ = null;
+
+  /**
+   * @type {?angular.$q.Promise|null}
+   * @private
+   */
+  this.rotationTimeoutPromise_ = null;
+
+  /**
+   * @return {ol.Size} Size in dots of the map to print.
+   */
+  var getSizeFn = function() {
+    return this.paperSize_;
+  }.bind(this);
+
+  /**
+   * @param {olx.FrameState} frameState Frame state.
+   * @return {number} Scale of the map to print.
+   */
+  var getScaleFn = function(frameState) {
+    var mapSize = frameState.size;
+    var viewResolution = frameState.viewState.resolution;
+    return this.fields.scale = this.getOptimalScale_(mapSize, viewResolution);
+  }.bind(this);
+
+  /**
+   * @return {number} rotation to apply.
+   */
+  var getRotationFn = function() {
+    return this.rotation;
+  }.bind(this);
+
+  /**
+   * @type {function(ol.render.Event)}
+   */
+  this.postcomposeListener_ = ngeoPrintUtils.createPrintMaskPostcompose(
+      getSizeFn, getScaleFn, getRotationFn);
+};
+
+
+/**
+ * @param {boolean} active True to listen events related to the print and get
+ *     capabilities. False to stop listen them and set rotation to 0.
+ * @private
+ */
+gmf.PrintController.prototype.togglePrintPanel_ = function(active) {
+  if (active) {
+    this.ngeoPrint_.getCapabilities().then(function(resp) {
+      this.printState = gmf.PrintState.NOT_IN_USE;
+      // Get capabilities - On success
+      this.parseCapabilities_(resp);
+      this.map.on('postcompose', this.postcomposeListener_);
+      this.map.on('pointerdrag', this.onPointerDrag_.bind(this));
+      this.map.render();
+    }.bind(this), function(resp) {
+      // Get capabilities - On error
+      this.printState = gmf.PrintState.ERROR_ON_GETCAPABILITIES;
+    }.bind(this));
+
+  } else {
+    this.map.un('postcompose', this.postcomposeListener_);
+    this.map.un('pointerdrag', this.onPointerDrag_.bind(this));
```

Comment:
Does this actually work? Calling `this.onPointerDrag_.bind(this)` generates a new function every time, so the `un` does not know what function you want to unregister. Better work with listener keys.